cmd/dogstatsd/main_windows.go

```diff
@@ -6,19 +6,370 @@
+		}
+	} else {
+		log.Debug("cmd_port not found, not setting")
+	}
+	for key, cfg := range subServices {
+		if val, _, err = k.GetStringValue(key); err == nil {
+			val = strings.ToLower(val)
+			if enabled, ok := enabledVals[val]; ok {
+				// some of the entries require booleans, some
+				// of the entries require strings.
+				if enabled {
+					switch cfg {
+					case "logs_enabled":
+						overrides[cfg] = true
+					case "apm_config.enabled":
+						overrides[cfg] = true
+					case "process_config.enabled":
+						overrides[cfg] = "true"
+					}
+					log.Debugf("Setting %s to true", cfg)
+				} else {
+					switch cfg {
+					case "logs_enabled":
+						overrides[cfg] = false
+					case "apm_config.enabled":
+						overrides[cfg] = false
+					case "process_config.enabled":
+						overrides[cfg] = "disabled"
+					}
+					log.Debugf("Setting %s to false", cfg)
+				}
+			} else {
+				log.Warnf("Unknown setting %s = %s", key, val)
+			}
+		}
+	}
+	if val, _, err = k.GetStringValue("proxy_host"); err == nil && val != "" {
+		var u *url.URL
+		if u, err = url.Parse(val); err != nil {
+			log.Warnf("unable to import value of settings 'proxy_host': %v", err)
+		} else {
+			// set scheme if missing
+			if u.Scheme == "" {
+				u, _ = url.Parse("http://" + val)
+			}
+			if val, _, err = k.GetStringValue("proxy_port"); err == nil && val != "" {
+				u.Host = u.Host + ":" + val
+			}
+			if user, _, _ := k.GetStringValue("proxy_user"); err == nil && user != "" {
+				if pass, _, _ := k.GetStringValue("proxy_password"); err == nil && pass != "" {
+					u.User = url.UserPassword(user, pass)
+				} else {
+					u.User = url.User(user)
+				}
+			}
+		}
+		proxyMap := make(map[string]string)
+		proxyMap["http"] = u.String()
+		proxyMap["https"] = u.String()
+		overrides["proxy"] = proxyMap
+	} else {
+		log.Debug("proxy key not found, not setting proxy config")
+	}
+	if val, _, err = k.GetStringValue("site"); err == nil && val != "" {
+		overrides["site"] = val
+		log.Debugf("Setting site to %s", val)
+	}
+	if val, _, err = k.GetStringValue("dd_url"); err == nil && val != "" {
+		overrides["dd_url"] = val
+		log.Debugf("Setting dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("logs_dd_url"); err == nil && val != "" {
+		overrides["logs_config.logs_dd_url"] = val
+		log.Debugf("Setting logs_config.dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("process_dd_url"); err == nil && val != "" {
+		overrides["process_config.process_dd_url"] = val
+		log.Debugf("Setting process_config.process_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("trace_dd_url"); err == nil && val != "" {
+		overrides["apm_config.apm_dd_url"] = val
+		log.Debugf("Setting apm_config.apm_dd_url to %s", val)
+	}
+	if val, _, err = k.GetStringValue("py_version"); err == nil && val != "" {
+		overrides["python_version"] = val
+		log.Debugf("Setting python version to %s", val)
+	}
+
+	// apply overrides to the config
+	config.AddOverrides(overrides)
```

Comment:
You probably already checked this, but I have to ask - are you sure the dots are dealt with properly with the overrides? ie. is `overrides['apm_config.apm_dd_url'] = val` overridden in the same way as `overrides['apm_config']['apm_dd_url'] = val` ?

Also wondering if it wouldn't be useful to have this overriding chunk of code tested to avoid issue moving forward and to help us identify bad assumptions.